{"info": {"author": "OpenNode Team", "author_email": "info@opennodecloud.com", "bugtrack_url": null, "classifiers": ["Framework :: Django", "Intended Audience :: Developers", "Intended Audience :: System Administrators", "License :: OSI Approved :: MIT License", "Operating System :: OS Independent"], "description": "Waldur Auth OpenID plugin\n=========================\n\nWaldur plugin bringing OpenID-based authentication support.\n\nUsage\n-----\n\n1. Install the plugin\n\n2. Run migrations to create new tables.\n\n.. code-block::\n\n    waldur migrate --noinput\n\n3. Add your OpenID to your user account:\n\n.. code-block::\n\n    Waldur admin > Core > Django_Openid_Auth > User open ids > Add user open id\n    User: <your-django-user-id>\n    Claimed id: <openid.claimed_id> (example: https://launchpad.net/~johndoe)\n    Display name: <your-display-name>\n\n4. Navigate to `<waldur-url>/api-auth/openid/login` and try to log in using your OpenID provider URL\n\nKnown issues\n------------\n\n**Database migrations do not work correctly with MySQL**\n\nRunning migrations (step 3 of 'Usage' section) may fail with following error:\n\n.. code-block::\n\n    django.db.utils.OperationalError: (1170, \"BLOB/TEXT column 'claimed_id' used in key specification without a key length\")\n\nThis is a `known issue<https://bugs.launchpad.net/django-openid-auth/+bug/524796>`_ of `django-openid-auth<https://pypi.python.org/pypi/django-openid-auth/>`_ :\nDjango `does not allow<https://code.djangoproject.com/ticket/2495>`_ `TextField` to be unique.\n\nFix exists but it is not yet merged as of 2016-05-03.\n\nTo work around the problem you will have to manually patch the `django-openid-auth` code as follows:\n\n.. code-block:: python\n\n    # File: django_openid_auth/migrations/0001_initial.py\n    -    claimed_id = models.TextField(max_length=2047, unique=True)\n    +    claimed_id = models.CharField(max_length=255, unique=True)\n\n    # File: django_openid_auth/migrations/0001_initial.py\n    -                ('claimed_id', models.TextField(unique=True, max_length=2047)),\n    +                ('claimed_id', models.CharField(unique=True, max_length=255)),\n\n-- and launch the migrations again.", "description_content_type": null, "docs_url": null, "download_url": "", "downloads": {"last_day": -1, "last_month": -1, "last_week": -1}, "home_page": "http://waldur.com", "keywords": "", "license": "MIT", "maintainer": "", "maintainer_email": "", "name": "waldur-auth-openid", "package_url": "https://pypi.org/project/waldur-auth-openid/", "platform": "", "project_url": "https://pypi.org/project/waldur-auth-openid/", "project_urls": {"Homepage": "http://waldur.com"}, "release_url": "https://pypi.org/project/waldur-auth-openid/0.8.9/", "requires_dist": null, "requires_python": "", "summary": "Waldur plugin bringing OpenID-based authentication support.", "version": "0.8.9", "yanked": false, "html_description": "<div class=\"project-description\">\n            Waldur Auth OpenID plugin<br>=========================<br><br>Waldur plugin bringing OpenID-based authentication support.<br><br>Usage<br>-----<br><br>1. Install the plugin<br><br>2. Run migrations to create new tables.<br><br>.. code-block::<br><br>    waldur migrate --noinput<br><br>3. Add your OpenID to your user account:<br><br>.. code-block::<br><br>    Waldur admin &gt; Core &gt; Django_Openid_Auth &gt; User open ids &gt; Add user open id<br>    User: &lt;your-django-user-id&gt;<br>    Claimed id: &lt;openid.claimed_id&gt; (example: https://launchpad.net/~johndoe)<br>    Display name: &lt;your-display-name&gt;<br><br>4. Navigate to `&lt;waldur-url&gt;/api-auth/openid/login` and try to log in using your OpenID provider URL<br><br>Known issues<br>------------<br><br>**Database migrations do not work correctly with MySQL**<br><br>Running migrations (step 3 of 'Usage' section) may fail with following error:<br><br>.. code-block::<br><br>    django.db.utils.OperationalError: (1170, \"BLOB/TEXT column 'claimed_id' used in key specification without a key length\")<br><br>This is a `known issue&lt;https://bugs.launchpad.net/django-openid-auth/+bug/524796&gt;`_ of `django-openid-auth&lt;https://pypi.python.org/pypi/django-openid-auth/&gt;`_ :<br>Django `does not allow&lt;https://code.djangoproject.com/ticket/2495&gt;`_ `TextField` to be unique.<br><br>Fix exists but it is not yet merged as of 2016-05-03.<br><br>To work around the problem you will have to manually patch the `django-openid-auth` code as follows:<br><br>.. code-block:: python<br><br>    # File: django_openid_auth/migrations/0001_initial.py<br>    -    claimed_id = models.TextField(max_length=2047, unique=True)<br>    +    claimed_id = models.CharField(max_length=255, unique=True)<br><br>    # File: django_openid_auth/migrations/0001_initial.py<br>    -                ('claimed_id', models.TextField(unique=True, max_length=2047)),<br>    +                ('claimed_id', models.CharField(unique=True, max_length=255)),<br><br>-- and launch the migrations again.\n          </div>"}, "last_serial": 3617979, "releases": {"0.8.5": [{"comment_text": "", "digests": {"md5": "f6e693db30c480700481942eb923c401", "sha256": "7ae8889bd19ea00a0873d60fb1868fbe4cb0c3cd81953351595dcd543d2a0527"}, "downloads": -1, "filename": "waldur_auth_openid-0.8.5-py2-none-any.whl", "has_sig": false, "md5_digest": "f6e693db30c480700481942eb923c401", "packagetype": "bdist_wheel", "python_version": "py2", "requires_python": null, "size": 8798, "upload_time": "2017-11-24T14:36:43", "upload_time_iso_8601": "2017-11-24T14:36:43.045489Z", "url": "https://files.pythonhosted.org/packages/5c/6f/7f61f678fd14195db3ac2cdf77be69167e0da496afa0cd9b2445686e64bd/waldur_auth_openid-0.8.5-py2-none-any.whl", "yanked": false}, {"comment_text": "", "digests": {"md5": "2eea0d5286b55e6521f3514186d4e954", "sha256": "43d12cc2c3c298645796baf1df867025d16318d91c154f177b26639d182e9e56"}, "downloads": -1, "filename": "waldur-auth-openid-0.8.5.tar.gz", "has_sig": false, "md5_digest": "2eea0d5286b55e6521f3514186d4e954", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 4913, "upload_time": "2017-11-24T14:36:44", "upload_time_iso_8601": "2017-11-24T14:36:44.875701Z", "url": "https://files.pythonhosted.org/packages/5d/36/dfae4cc478f7d80b0741b985d82255c5b032d244da23696b260df8394587/waldur-auth-openid-0.8.5.tar.gz", "yanked": false}], "0.8.6": [{"comment_text": "", "digests": {"md5": "7b3a0da28cfbdf17b899b82a61662a2f", "sha256": "fd646bb6c3dbb38caff9d946657d0b7024b105a06fdd67e4da82f7ab081bd063"}, "downloads": -1, "filename": "waldur-auth-openid-0.8.6.tar.gz", "has_sig": false, "md5_digest": "7b3a0da28cfbdf17b899b82a61662a2f", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 4914, "upload_time": "2017-11-27T08:54:39", "upload_time_iso_8601": "2017-11-27T08:54:39.354420Z", "url": "https://files.pythonhosted.org/packages/e1/39/66f816e6deec55fd4d462896ec88dca9e24c69bc62ddf33923f6e8358c5c/waldur-auth-openid-0.8.6.tar.gz", "yanked": false}], "0.8.7": [{"comment_text": "", "digests": {"md5": "10951cc4f11ae87faf9056632c96a561", "sha256": "9105750f7d44535622d6b8331e60900bf503eeae191ac06f3cd43e9fb6ce06cf"}, "downloads": -1, "filename": "waldur-auth-openid-0.8.7.tar.gz", "has_sig": false, "md5_digest": "10951cc4f11ae87faf9056632c96a561", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 4906, "upload_time": "2017-12-01T13:01:11", "upload_time_iso_8601": "2017-12-01T13:01:11.725594Z", "url": "https://files.pythonhosted.org/packages/94/d2/2dc66e6f6442be0508a9a80497d5c96bc0de42c46ba76934632ba39d2c22/waldur-auth-openid-0.8.7.tar.gz", "yanked": false}], "0.8.8": [{"comment_text": "", "digests": {"md5": "2f16f5096626f0477f385db680c3a150", "sha256": "740ee11e7362d2e5330c649e743bc4be57dab4da88be7c89e8a47cf9b8862dea"}, "downloads": -1, "filename": "waldur-auth-openid-0.8.8.tar.gz", "has_sig": false, "md5_digest": "2f16f5096626f0477f385db680c3a150", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 4777, "upload_time": "2018-02-12T17:57:17", "upload_time_iso_8601": "2018-02-12T17:57:17.656354Z", "url": "https://files.pythonhosted.org/packages/ca/8a/033b0c4c05ee7a3b5ec31ce3b075fc47d831f8e1ec070a9f36308a924202/waldur-auth-openid-0.8.8.tar.gz", "yanked": false}], "0.8.9": [{"comment_text": "", "digests": {"md5": "c8fb0c54551c87b5637877630e22bf6e", "sha256": "0b573d9cdd4717f7a70f1ab8e8d5fcf15a55048b3d6680d9ffaf407a74889ea8"}, "downloads": -1, "filename": "waldur-auth-openid-0.8.9.tar.gz", "has_sig": false, "md5_digest": "c8fb0c54551c87b5637877630e22bf6e", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 4878, "upload_time": "2018-02-26T16:41:15", "upload_time_iso_8601": "2018-02-26T16:41:15.884613Z", "url": "https://files.pythonhosted.org/packages/6e/cd/ae72c94e803f6f28aa6b897992525899e37f94a73c67d07465f8abb3d12d/waldur-auth-openid-0.8.9.tar.gz", "yanked": false}]}, "urls": [{"comment_text": "", "digests": {"md5": "c8fb0c54551c87b5637877630e22bf6e", "sha256": "0b573d9cdd4717f7a70f1ab8e8d5fcf15a55048b3d6680d9ffaf407a74889ea8"}, "downloads": -1, "filename": "waldur-auth-openid-0.8.9.tar.gz", "has_sig": false, "md5_digest": "c8fb0c54551c87b5637877630e22bf6e", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 4878, "upload_time": "2018-02-26T16:41:15", "upload_time_iso_8601": "2018-02-26T16:41:15.884613Z", "url": "https://files.pythonhosted.org/packages/6e/cd/ae72c94e803f6f28aa6b897992525899e37f94a73c67d07465f8abb3d12d/waldur-auth-openid-0.8.9.tar.gz", "yanked": false}], "timestamp": "Fri May  8 03:32:42 2020"}